plugins/modules/foreman_job_template.py

```diff
@@ -388,72 +380,66 @@ def main():
     affects_multiple = name == '*'
     # sanitize user input, filter unuseful configuration combinations with 'name: *'
     if affects_multiple:
-        if state == 'present_with_defaults':
+        if module.state == 'present_with_defaults':
             module.fail_json(msg="'state: present_with_defaults' and 'name: *' cannot be used together")
-        if state == 'absent':
+        if module.state == 'absent':
```

Comment:
```suggestion
        if module.desired_absent:
```

Even tho I am not sure the check is *really* useful ;)